app/src/main/java/org/wikipedia/settings/languages/WikipediaLanguagesItemView.kt

```diff
@@ -0,0 +1,92 @@
+package org.wikipedia.settings.languages
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.os.Build
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import android.widget.LinearLayout
+import androidx.appcompat.content.res.AppCompatResources
+import androidx.core.graphics.BlendModeColorFilterCompat
+import androidx.core.graphics.BlendModeCompat
+import org.apache.commons.lang3.StringUtils
+import org.wikipedia.R
+import org.wikipedia.databinding.ItemWikipediaLanguageBinding
+import org.wikipedia.search.SearchFragment
+import org.wikipedia.util.DeviceUtil.setContextClickAsLongClick
+import org.wikipedia.util.ResourceUtil.getThemedAttributeId
+import org.wikipedia.util.ResourceUtil.getThemedColor
+import org.wikipedia.views.ViewUtil.formatLangButton
+
+class WikipediaLanguagesItemView : LinearLayout {
+    interface Callback {
+        fun onCheckedChanged(position: Int)
+        fun onLongPress(position: Int)
+    }
+
+    private var binding = ItemWikipediaLanguageBinding.inflate(LayoutInflater.from(context), this)
+    private var position = 0
+    var callback: Callback? = null
+
+    constructor(context: Context?) : super(context)
+    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs)
+    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr)
+
+    init {
+        layoutParams = ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT,
+                ViewGroup.LayoutParams.WRAP_CONTENT)
+        setBackgroundColor(getThemedColor(context, R.attr.paper_color))
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            foreground = AppCompatResources.getDrawable(context,
+                    getThemedAttributeId(context, R.attr.selectableItemBackground))
+        }
+        binding.wikiLanguageCheckbox.setOnCheckedChangeListener { _, _ ->
+            callback?.onCheckedChanged(position)
+            updateBackgroundColor()
+        }
+        setOnLongClickListener {
+            callback?.onLongPress(position)
+            true
+        }
+        setContextClickAsLongClick(this)
+    }
+
+    private fun updateBackgroundColor() {
+        setBackgroundColor(if (binding.wikiLanguageCheckbox.isChecked) getThemedColor(context, R.attr.multi_select_background_color) else getThemedColor(context, R.attr.paper_color))
+    }
+
+    fun setContents(langCode: String, languageLocalizedName: String?, position: Int) {
+        this.position = position
+        binding.wikiLanguageOrder.text = (position + 1).toString()
+        binding.wikiLanguageTitle.text = StringUtils.capitalize(languageLocalizedName)
```

Comment:
Can be replaced with `orEmpty().capitalize(Locale.getDefault)`

Reply:
Done. Thanks.